plugins/modules/packaging/language/maven_artifact.py

```diff
@@ -150,6 +150,11 @@
         default: 'md5'
         choices: ['md5', 'sha1']
         version_added: 3.2.0
+    unredirected_headers:
+        type: list
+        description:
+            A list of headers that should not be included in the redirection. This headers are sent to the fetch_url module.
+            Useful if the redirection URL does not need to have sensitive headers in the request.
```

Comment:
It would probably make sense to add Authorization and Cookies as the default

Reply:
I don't think that's a good idea, since a) it will break all ansible-core 2.11 users (if you don't add detection code for them), and b) this is a potential security problem since usually such headers are not included in redirects intentionally. Not having a default (or the default being an empty list) prevents this.

Reply:
> ... b) this is a potential security problem since usually such headers are not included in redirects intentionally. Not having a default (or the default being an empty list) prevents this.

Authorization and Cookie are passed by default. This is exactly the problem that is aimed to be solved. See:
- https://github.com/python/cpython/issues/77842
- https://github.com/python/cpython/pull/11292

Reply:
Ah sorry, I think I missed the **un** in the option name :)

In any case, having a default here means there **must** be code that detects whether this option can actually be passed to fetch_url.